### PR TITLE
ci: freeze requirements used in workflows

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

some are frozen without visibility, and others are without any versions
the motivation was seeing broken uploads with `awscli` with latest versions

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19146.org.readthedocs.build/en/19146/

cc @borda @carmocca

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

**Review by lantiga — APPROVED** [review]

File: .github/workflows/ci-pkg-install.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/*"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-pkg-install.yml:12]: `requirements/**` is already listed in L19

File: .github/workflows/ci-tests-app.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ on:
     paths:
       - ".actions/*"
       - ".github/workflows/ci-tests-app.yml"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-tests-app.yml:13]: This change should be reflected on `.github/checkgroup.yml`

File: .github/workflows/ci-tests-data.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/*"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-tests-data.yml:12]: This change should be reflected on `.github/checkgroup.yml`

File: .github/workflows/ci-tests-fabric.yml [hunk 4]
```diff
@@ -9,6 +9,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/*"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-tests-fabric.yml:12]: This change should be reflected on `.github/checkgroup.yml`

File: .github/workflows/ci-tests-pytorch.yml [hunk 5]
```diff
@@ -9,6 +9,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/*"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-tests-pytorch.yml:12]: This change should be reflected on `.github/checkgroup.yml`

File: .github/workflows/ci-tests-store.yml [hunk 6]
```diff
@@ -9,6 +9,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize] # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/*"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/ci-tests-store.yml:12]: This change should be reflected on `.github/checkgroup.yml`

File: .github/workflows/docs-build.yml [hunk 7]
```diff
@@ -11,6 +11,7 @@ on:
     paths:
       - ".actions/*"
       - ".github/workflows/docs-build.yml"
+      - "requirements/ci.txt"
```
Comment (carmocca) [inline .github/workflows/docs-build.yml:14]: Here too

**Review by carmocca — COMMENTED** [review]
Friendly reminder that job paths should match checkgroup paths

**Comment by Borda** [comment]
> Friendly reminder that job paths should match checkgroup paths

I don't think this need to be required

**Comment by carmocca** [comment]
If they don't match, the following can happen:

1. `ci.txt` is updated to include a new release of a dependency
2. that release doesn't work (the PR CI is red)
3. checkgroup won't require those jobs because it doesn't know about `ci.txt`
4. the PR can be merged with failing CI, which would block other PRs.

We don't allow auto-merge anymore, so it wouldn't happen automatically. But if checkgroup is not updated then this is a way to break CI from user error

**Comment by Borda** [comment]
> 1. `ci.txt` is updated to include a new release of a dependency

That is the purpose so we can control it 

> 2. that release doesn't work (the PR CI is red)

True, as it would be also before this PR... 

> 3. checkgroup won't require those jobs because it doesn't know about `ci.txt`

Why would you merge anything to break master? 

> 4. the PR can be merged with failing CI, which would block other PRs.

True and False at the same time...

> We don't allow auto-merge anymore, so it wouldn't happen automatically. But if checkgroup is not updated then this is a way to break CI from user error

Happy that you agree that this PR brings only important to state before =)

**Comment by carmocca** [comment]
This PR is great. I'm just pointing out things that should be **added**.

**Comment by Borda** [comment]
> This PR is great. I'm just pointing out things that should be **added**.

it did the main job, all others can be added later as you suggested to make rather smaller changes

**Comment by carmocca** [comment]
I'm glad that you agree. Will you be able to open a PR making them?

**Pull request merged**